### PR TITLE
PR: Add validation prehooks

### LINKED ISSUE #16: Add cookiecutter argument validation with a pre hook python script
http://cookiecutter.readthedocs.io/en/latest/advanced/hooks.html#example-validating-template-variables

### PR DESCRIPTION
**Pull request by rlaverde**
Fixes: #16

### CODE REVIEW

File: hooks/pre_gen_project.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+project_name = '{{ cookiecutter.project_name }}'
+
+if hasattr(project_name, 'isidentifier'):
+    assert project_name.isidentifier(), 'Project slug should be valid Python identifier!'
+
+assert project_name.startswith('spyder'), 'Project slug should start with "spyder".'
```
Comment (goanpeca) [inline hooks/pre_gen_project.py]: Lets use regex

File: hooks/pre_gen_project.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+project_name = '{{ cookiecutter.project_name }}'
+
+if hasattr(project_name, 'isidentifier'):
```
Comment (goanpeca) [inline hooks/pre_gen_project.py]: I dont understand this :-|

**Review by goanpeca — CHANGES_REQUESTED** [review]

**Comment by goanpeca** [comment]
@rlaverde no need to add labels to PRs

**Comment by rlaverde** [comment]
@goanpeca merge?

File: hooks/pre_gen_project.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import re
+import sys
+
+PROJECT_REGEX = r'^spyder_[_a-zA-Z0-9]+$'
```
Comment (goanpeca) [inline hooks/pre_gen_project.py]: it should be all lowercase, so A-Z is not ok

File: hooks/pre_gen_project.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+project_name = '{{ cookiecutter.project_name }}'
+
+if not re.match(PROJECT_REGEX, project_name):
+    print('ERROR: {}'.format(project_name),
+          'Project name should start with "spyder"',
+          'and be a valid Python module!')
+
```
Comment (goanpeca) [inline hooks/pre_gen_project.py:12]: We should also validate the name of the repo or plugin

it should be 

Spyder-Something

or spyder-something

**Review by goanpeca — CHANGES_REQUESTED** [review]

**Comment by rlaverde** [comment]
As discussed by chat I only add validation for plugin_name, and I'll open another PR to make other variables invisible.

EDIT: Making other variables invisible isn't easy (See: #7) we should validate module name instead

**Pull request closed without merging**